### PR TITLE
fix: use firebase access token directly instead of redux saved token

### PR DESCRIPTION
**Pull request by davidgamez**
**Summary:**

This PR addresses using the user's access token while requesting the feed's API endpoints. Principles considered for using the Firebase SDK function instead of the redux state for the API token management:

- Previously the access token used is the one saved in the redux store. This token was initially saved in redux as part of the account screen. Using this token can lead to cases where the token is already created before the user hits the account screen, having the code to manage different entry stages to the create access token button. In addition, access tokens expire every hour, so we will need to handle the rotation of the token and the implications of this process on the current account's page. There is a potential race condition when updating a token to the redux state, and the user requests an API call with an already expired token.
- The PR uses the token _managed_ by the Firebase SDK directly. The firebase already caches the token and is able to request a new one if it expires. The additional benefit is that the access token management is also abstracted from the UI component, leaving the service layer to perform token rotation when needed.

Part of #516 

## Out of scope

- This PR doesn't address the persistent of the redux state across tabs
- Cypress test will be added in a separate PR

**Expected behavior:** 

The search and feed screen shows the correct information for anonymous and authenticated users.

**Testing tips:**

Anonymous user:
- Open the App in the home screen
- Click "Browse Feeds"
- Expected feed results to render on the screen
- Click on a feed and verify that all information is presented
- Wait an +hour(do not refresh the page)
- Click back and verify that feed results are rendered

Authenticated user:
- Open the App in the home screen
- Login to the app with any method(please try multiple one at a time)
- Click the "Feeds" link from the top header menu
- Expected feed results to render on the screen

- Try also different combinations of login and logout operations

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: web-app/src/app/store/saga/dataset-saga.ts [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
-  paths['/v1/gtfs_feeds/{id}/datasets']['get']['responses'][200]['content']['application/json']
-> {
+  payload: { feedId },
+}: PayloadAction<{ feedId: string }>): Generator {
   try {
     if (feedId !== undefined) {
-      const datasets = yield call(getGtfsFeedDatasets, feedId, accessToken, {});
+      const accessToken: string | null = (yield call(
+        getUserAccessToken,
+      )) as string;
```
Comment (Alessandro100) [inline web-app/src/app/store/saga/dataset-saga.ts:18]: we don't need to declare `string | null` and `as string`

File: web-app/src/app/store/saga/feed-saga.ts [hunk 2]
```diff
@@ -20,16 +20,20 @@ import {
...
+}: PayloadAction<{ feedId: string }>): Generator<
   StrictEffect,
   void,
   AllFeedType
 > {
   try {
     if (feedId !== undefined) {
+      const accessToken: string | null = (yield call(
+        getUserAccessToken,
+      )) as string;
```
Comment (Alessandro100) [inline web-app/src/app/store/saga/feed-saga.ts:36]: same comment

File: web-app/src/app/store/saga/feed-saga.ts [hunk 3]
```diff
@@ -43,15 +47,14 @@ function* getFeedSaga({
...
-  AllFeedType[]
-> {
+  payload: { feedIds },
+}: PayloadAction<{ feedIds: string[] }>): Generator {
   try {
     if (feedIds.length > 0) {
-      const feedsData: AllFeedType[] = yield all(
+      const accessToken: string | null = (yield call(
+        getUserAccessToken,
+      )) as string;
```
Comment (Alessandro100) [inline web-app/src/app/store/saga/feed-saga.ts:56]: same

File: web-app/src/app/services/profile-service.ts [hunk 4]
```diff
@@ -35,6 +35,22 @@ export const getUserFromSession = async (): Promise<User | null> => {
...
+export async function getUserAccessToken(): Promise<string> {
+  const currentUser = app.auth().currentUser;
+
+  if (currentUser === null) {
+    throw new Error('Cannot retrieve access token');
+  }
+  const idTokenResult = await currentUser.getIdTokenResult();
+
+  return idTokenResult.token;
+}
```
Comment (Alessandro100) [inline web-app/src/app/services/profile-service.ts:52]: this is great! Handles a decent amount of auth info for us 🎉

**Review by Alessandro100 — APPROVED** [review]
left a few small comments otherwise it works great ✅

File: web-app/src/app/store/saga/dataset-saga.ts (hunk 1 above)
Comment (davidgamez) [reply web-app/src/app/store/saga/dataset-saga.ts:18]: Good observation. However, it's actually needed at list the `as string`. The reason is that generators functions don't play nice with TS with multiple calls. The root cause is that the generators functions "always" returns any. we were enforcing the type discovery defined  StriictEffect type but this doesn't account for multiple calls. The way I was able to go around is to define the return type a generic `Generator` vs the previous pattern was like `Generator<StrictEffect, void,AllFeedType>`.  
In any case, you are right I can skip the `string | null` definition keeping the `as string`, resulting code `const accessToken = (yield call(getUserAccessToken)) as string;`

**Review by qcdyx — APPROVED** [review]
LGTM

**Comment by davidgamez** [comment]
> left a few small comments otherwise it works great ✅

Thanks!! addressed your feedback!

**Pull request merged**